raft/raft.go

```diff
@@ -654,6 +671,11 @@ func stepLeader(r *raft, m pb.Message) {
 					// an update before, send it now.
 					r.sendAppend(m.From)
 				}
+				// Transfer leadership is in progress.
+				if m.From == r.leadTransferee && pr.Match == r.raftLog.lastIndex() && !r.timeoutNowSent {
+					r.logger.Infof("%x sent MsgTimeoutNow to %x after received MsgAppResp", r.id, m.From)
```

Comment:
If we assume that the transport (usually) preserves order, we could send the `MsgTimeoutNow` as soon as we have sent a `MsgApp` containing the last index, without waiting for the `MsgAppResp`. This is similar to the way we optimistically increase `Progress.Next` when sending `MsgApp`.